tool/tsh/tsh.go

```diff
@@ -3391,6 +3391,19 @@ func makeClientForProxy(cf *CLIConf, proxy string) (*client.TeleportClient, erro
 		}
 	}
 
+	// When using Headless, check for missing proxy/user/cluster values from the teleport session env variables.
+	if cf.AuthConnector == constants.HeadlessConnector {
```

Comment:
Why do we check for headless auth connector here and not `cf.Headless` as above? Even if `--headless` is a shorthand for `--auth=headless`, we should only have one way of performing this check.

Reply:
In the check above, we use `cf.Headless` to set `cf.AuthConnector` so that below we can just check `cf.AuthConnector`.